server/src/api/http/auth.py

```diff
@@ -123,3 +131,87 @@ async def reset_password(request):
     user.save()
 
     return web.HTTPOk()
+
+# OIDC Authentication endpoints
+
+async def oidc_providers(request):
+    if "oidc" not in cfg().general.authentication_methods:
+        return web.HTTPForbidden(reason="OIDC authentication is disabled")
+    providers = oidc_auth.get_providers()
+    return web.json_response({"providers": providers})
+
+async def oidc_login(request):
+    logger = logging.getLogger("PlanarAllyServer")
+    if "oidc" not in cfg().general.authentication_methods:
+        logger.warning(f"OIDC authentication attempted but disabled in config")
+        return web.HTTPForbidden(reason="OIDC authentication is disabled")
+    try:
+        data = await request.json()
+    except Exception as e:
+        return web.HTTPBadRequest(reason="Invalid request format")
+    
+    # Grab the state from the browser's request ensuring we can validate it later
+    provider_name = data.get("provider_name")
+    if not provider_name:
+        return web.HTTPBadRequest(reason="Missing required parameters")
+
+    auth_url = await oidc_auth.get_authorization_url(provider_name)
+
+    if not auth_url:
+        return web.HTTPInternalServerError(reason="Failed to initiate OIDC login")
+    logger.debug(f"Redirecting to OIDC provider with URL: {auth_url}")
```

Comment:
I don't think adding the full auth_url is a good idea, won't it include sensitive info?